src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -763,6 +768,15 @@ public RegisteredUserDTO createUserObjectAndSession(final HttpServletRequest req
             throw new MissingRequiredFieldException("The email address provided is invalid.");
         }
 
+        // validate names
+        if (!this.isUserNameValid(user.getGivenName())) {
+            throw new InvalidNameException("The given name provided is an invalid length or contains illegal characters.");
```

Comment:
I think "forbidden characters" might have been more clear to a child than "illegal" which has other connotations outside Computer Science disciplines!
Also, unless we want to provide feedback about what specifically is wrong with the name, this message could be in a default constructor in the `InvalidNameException` class so it wasn't repeated.

Reply:
Given name/family name aligns with the property names in the request, so I think this is a sensible response from the backend. The frontend will need to change to handle this consistently anyway (it currently also validates length but not special characters).